### PR TITLE
LPS-183098 Infer template language instead of assuming velocity

### PR DESCRIPTION
**Pull request by izaera**

### CODE REVIEW

File: portal-kernel/src/com/liferay/portal/kernel/layoutconfiguration/util/RuntimePage.java [hunk 1]
```diff
@@ -30,7 +30,7 @@
 @ProviderType
 public interface RuntimePage {
 
-	public LayoutTemplate getLayoutTemplate(String velocityTemplateId);
+	public LayoutTemplate getLayoutTemplate(String templateId);
```
Comment (dsanz) [inline portal-kernel/src/com/liferay/portal/kernel/layoutconfiguration/util/RuntimePage.java:33]: We shall change this in [RuntimePageUtil](https://github.com/liferay/liferay-portal/blob/0a9951c25d434cc515f620ffba68c3a9e4fbaeed/portal-kernel/src/com/liferay/portal/kernel/layoutconfiguration/util/RuntimePageUtil.java#L30) as well

File: portal-kernel/src/com/liferay/portal/kernel/layoutconfiguration/util/RuntimePageUtil.java [hunk 2]
```diff
@@ -27,8 +27,8 @@
  */
 public class RuntimePageUtil {
 
-	public static LayoutTemplate getLayoutTemplate(String velocityTemplateId) {
-		return _runtimePage.getLayoutTemplate(velocityTemplateId);
+	public static LayoutTemplate getLayoutTemplate(String templateId) {
+		return _runtimePage.getLayoutTemplate(templateId);
```
Comment (dsanz) [inline portal-kernel/src/com/liferay/portal/kernel/layoutconfiguration/util/RuntimePageUtil.java:31]: Blazing fast!!!

Comment (izaera) [reply portal-kernel/src/com/liferay/portal/kernel/layoutconfiguration/util/RuntimePageUtil.java:31]: #RealTimeFix

File: portal-impl/src/com/liferay/portal/layoutconfiguration/util/RuntimePageImpl.java [hunk 3]
```diff
@@ -142,9 +109,23 @@ protected StringBundler doDispatch(
 			String templateId, String content, String langType)
 		throws Exception {
 
+		LayoutTemplateLocator layoutTemplateLocator = new LayoutTemplateLocator(
+			templateId);
+
+		if (langType == null) {
+			langType = LayoutTemplateLocalServiceUtil.getLangType(
```
Comment (dsanz) [inline portal-impl/src/com/liferay/portal/layoutconfiguration/util/RuntimePageImpl.java:116]: It was there in front of our eyes, nice!

**Comment by dsanz** [comment]
So, no breaking change and no need to change portal usages :partying_face:

File: portal-impl/src/com/liferay/portal/layoutconfiguration/util/RuntimePageImpl.java (hunk 3 above)
Comment (izaera) [reply portal-impl/src/com/liferay/portal/layoutconfiguration/util/RuntimePageImpl.java:116]: Yeah, that method saved our lives :sweat_smile:

**Comment by eltonccdantas** [comment]
Hey @izaera, I've analyzed your pull request. Although there are a few failures, the only one that seems to be related to it is the one below: 


**Info Framework | com.liferay.info.request.struts.test.AddInfoItemStrutsActionTest**

Error message:

2023-05-10 15:56:39.611 WARN  [com.liferay.info.test-executor-thread][RawMetadataProcessorMessageListener:43] Unable to save metadata for file version 61800
com.liferay.portal.kernel.exception.SystemException: java.io.IOException: org.apache.tika.exception.TikaException: TesseractOCRParser bad exit value 1 err msg: Tesseract Open Source OCR Engine v3.04.00 with Leptonica
read_params_file: Can't open 1
read_params_file: Can't open c
read_params_file: Can't open page_separator=
read_params_file: Can't open c
read_params_file: Can't open preserve_interword_spaces=0
read_params_file: Can't open txt
Error in fopenReadStream: file not found
Error in pixRead: image file not found: P72elG7q
Image file P72elG7q cannot be read!
Error during processing.

	at com.liferay.portal.tika.internal.metadata.TikaRawMetadataProcessor._extractMetadata(TikaRawMetadataProcessor.java:242) ~[?:?]
	at com.liferay.portal.tika.internal.metadata.TikaRawMetadataProcessor.getRawMetadataMap(TikaRawMetadataProcessor.java:99) ~[?:?]
	at com.liferay.portal.kernel.metadata.RawMetadataProcessorUtil.getRawMetadataMap(RawMetadataProcessorUtil.java:39) ~[portal-kernel.jar:?]
	at com.liferay.portlet.documentlibrary.util.RawMetadataProcessorImpl.saveMetadata(RawMetadataProcessorImpl.java:133) ~[portal-impl.jar:?]
	at com.liferay.document.library.kernel.util.RawMetadataProcessorUtil.saveMetadata(RawMetadataProcessorUtil.java:106) ~[portal-kernel.jar:?]
	at com.liferay.portlet.documentlibrary.messaging.RawMetadataProcessorMessageListener.doReceive(RawMetadataProcessorMessageListener.java:39) ~[portal-impl.jar:?]
	at com.liferay.portal.kernel.messaging.BaseMessageListener.doReceive(BaseMessageListener.java:48) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.messaging.BaseMessageListener.receive(BaseMessageListener.java:34) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.messaging.InvokerMessageListener.receive(InvokerMessageListener.java:74) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.test.rule.SynchronousDestinationTestRule$TestSynchronousDestination.send(SynchronousDestinationTestRule.java:412) ~[?:?]



Also, I  don't know if it is the cause of:     
` [exec] SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".`

or if it is caused by it...

What do you think?

[Comparison Run](https://testray.liferay.com/home/-/testray/runs/compare?compare=true&view=details&statusA=2&testrayRunIdB=2300262217&testrayRunIdA=2299637243&statusB=3)

[Testray-Result](https://github.com/liferay-frontend/liferay-portal/files/11451226/testray-results.csv)

**Comment by eltonccdantas** [comment]
ci:test:relevant

**Comment by izaera** [comment]
ci:forward:force

**Comment by izaera** [comment]
ci:forward:force

**Pull request closed without merging**